### PR TITLE
Drone: Add Slack failure notifier

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Add Slack failure notifier to Drone. Not added for PR builds for now, since the Slack webhook could easily be leaked, until we protect the Drone configuration.

### CODE REVIEW

File: scripts/lib.star [hunk 1]
```diff
@@ -286,6 +315,11 @@ def init_steps(edition, platform, is_downstream=False, install_deps=True):
         },
     ]
 
+    if version_mode == 'master':
+        steps.append(slack_step(channel='grafana-ci-failures'))
```
Comment (marefr) [inline scripts/lib.star:319]: I've already setup a webhook secret for the #grafana-ci slack channel. If you rather would like to use this one let me know and I'll delete my channel and webhook token

**Review by marefr — APPROVED** [review]
Great. Started working on this, but other things came in between so never opened any PR. LGTM, but please see comment

File: scripts/lib.star (hunk 1 above)
Comment (aknuds1) [reply scripts/lib.star:319]: @marefr Aha... I saw that secret and got confused where it came from, so I replaced it with my own. What channel had you set up already?

Comment (aknuds1) [reply scripts/lib.star:319]: Merging now, since we can discuss which webhook to use separately :)

**Pull request merged**